tests/scm/BasicSCMUTest.cxxtest

```diff
@@ -287,6 +290,50 @@ void BasicSCMUTest::test_truth_value(void)
 
 // ============================================================
 
+void BasicSCMUTest::test_values(void)
+{
+	logger().debug("BEGIN TEST: %s", __FUNCTION__);
+
+	eval->eval("(define h (Concept \"foo\"))");
+	TSM_ASSERT("Failed to create atom", not eval->eval_error());
+	eval->eval("(define k (Predicate \"key\"))");
+	TSM_ASSERT("Failed to create key", not eval->eval_error());
+	eval->eval("(define f (FloatValue 0.1 0.2 0.3))");
+	TSM_ASSERT("Failed to create float", not eval->eval_error());
+	eval->eval("(define s (StringValue \"a\" \"bb\" \"ccc\"))");
+	TSM_ASSERT("Failed to create string", not eval->eval_error());
+	eval->eval("(define t (stv 0.5 0.8))");
+	TSM_ASSERT("Failed to create TV", not eval->eval_error());
+	eval->eval("(define l (LinkValue f h s k t))");
+	TSM_ASSERT("Failed to create string", not eval->eval_error());
+
+	Handle h = eval->eval_h("h");
+	Handle k = eval->eval_h("k");
+	ProtoAtomPtr l = eval->eval_v("l");
+
+	eval->eval("(cog-set-value! h k l)");
+	TSM_ASSERT("Failed to set value", not eval->eval_error());
+
+	ProtoAtomPtr lv = eval->eval_v("(cog-value h k)");
+	TSM_ASSERT("Failed to get value", not eval->eval_error());
+	TSM_ASSERT("Failed to get correct value", l == lv);
+
+	ProtoAtomPtr alv = h->getValue(k);
+	TSM_ASSERT("Failed to get correct value", l == alv);
+
+	eval->eval("(cog-set-value! h k '(0.1 0.2 0.3))");
+	TSM_ASSERT("Failed to set float", not eval->eval_error());
```

Comment:
what you want is to say  `(cog-set-value! h k (list f s))`  -- and that will work.

the tick mark ' actually means "quote"  not "list" ... I'm trying to figure out if this is a user-error or a bug...